analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/Binary.cs

```diff
@@ -238,4 +257,10 @@ kind switch
                 _ => null
             }
             : null;
+
+    // Fixed loops:
+    // 1st visit decides on the initial value. We don't learn from binary comparison.
+    // 2nd visit does not decide on the current value. It learns range from binary comparison instead to be able to exit the loop.
+    private static bool IsFirstLoopVisit(int visitCount) =>
+        visitCount == 1;
```

Comment:
Why did you decide to make this a method?

Reply:
There's a hidden dependency between those two invocation sides - that are on two different logical paths. This extraction is making sure that they are structurally synchronized.
The fact that they need to be in sync is hard to see otherwise.

It's not very nice, because it's trivial. But makes sure it doesn't go out of sync. And it might change in the #7234 .